### PR TITLE
Made everything but tests/mp compile w/o boost

### PR DESCRIPTION
**Pull request by neubig**
Adds a new `INCLUDE_BOOST` definition for cmake, which is disabled by default. Tests and MP are only compiled if it is enabled.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -75,7 +75,7 @@ before_script:
   - cd build
   - conda create -q -n dynet-environment python=$PYVER $CONDA_PACKAGES
   - source activate dynet-environment
-  - cmake .. -DEIGEN3_INCLUDE_DIR=/usr/local/include/eigen3 -DPYTHON=`which python`
+  - cmake .. -DEIGEN3_INCLUDE_DIR=/usr/local/include/eigen3 -DINCLUDE_BOOST -DPYTHON=`which python`
```
Comment (xunzhang) [inline .travis.yml]: Use `-DINCLUDE_BOOST=ON` instead, `ON` can also be `On` or `on`.

File: CMakeLists.txt [hunk 2]
```diff
@@ -179,4 +177,7 @@ if(INCLUDE_SWIG)
   add_subdirectory(contrib/swig)
 endif(INCLUDE_SWIG)
 
-enable_testing()
+if(INCLUDE_BOOST)
```
Comment (xunzhang) [inline CMakeLists.txt]: I like this! 😎

File: dynet/str-util.h [hunk 3]
```diff
@@ -6,16 +6,16 @@
 
 namespace dynet {
```
Comment (xunzhang) [inline dynet/str-util.h]: I'll follow this file naming style in the future.

**Comment by neubig** [comment]
Thanks for the review!

**Pull request merged**